t/Local.t

```diff
@@ -4,304 +4,304 @@ use strict;
+            # other testers should get 2
+            ok( $hour == 2 || $hour == 3, 'hour should be 2 or 3' );
         }
+    },
+);
 
-        # Test timegm()
-        {
-            my $year_in = $year < 70 ? $year + 1900 : $year;
-            my $time = timegm( $sec, $min, $hour, $mday, $mon, $year_in );
-
-            my ( $s, $m, $h, $D, $M, $Y ) = gmtime($time);
-
-            is( $s, int($sec), "timegm second for @$_" );
-            is( $m, $min,      "timegm minute for @$_" );
-            is( $h, $hour,     "timegm hour for @$_" );
-            is( $D, $mday,     "timegm day for @$_" );
-            is( $M, $mon,      "timegm month for @$_" );
-            is( $Y, $year,     "timegm year for @$_" );
+subtest(
+    'Time::Local::_is_leap_year',
+    sub {
+        my @years = (
+            [ 1900 => 0 ],
+            [ 1947 => 0 ],
+            [ 1996 => 1 ],
+            [ 2000 => 1 ],
+            [ 2100 => 0 ],
+        );
+
+        for my $p (@years) {
+            my ( $year, $is_leap_year ) = @$p;
+
+            my $string = $is_leap_year ? 'is' : 'is not';
+            ## no critic (Subroutines::ProtectPrivateSubs)
+            is(
+                Time::Local::_is_leap_year($year), $is_leap_year,
+                "$year $string a leap year"
+            );
         }
     }
-}
-
-for (@bad_time) {
-    my ( $year, $mon, $mday, $hour, $min, $sec ) = @$_;
-    $year -= 1900;
-    $mon--;
-
-    eval { timegm( $sec, $min, $hour, $mday, $mon, $year ) };
-
-    like( $@, qr/.*out of range.*/, 'invalid time caused an error' );
-}
-
-{
-    is(
-        timelocal( 0, 0, 1, 1, 0, 90 ) - timelocal( 0, 0, 0, 1, 0, 90 ), 3600,
-        'one hour difference between two calls to timelocal'
-    );
-
-    is(
-        timelocal( 1, 2, 3, 1, 0, 100 ) - timelocal( 1, 2, 3, 31, 11, 99 ),
-        24 * 3600,
-        'one day difference between two calls to timelocal'
-    );
-
-    # Diff beween Jan 1, 1980 and Mar 1, 1980 = (31 + 29 = 60 days)
-    is(
-        timegm( 0, 0, 0, 1, 2, 80 ) - timegm( 0, 0, 0, 1, 0, 80 ),
-        60 * 24 * 3600,
-        '60 day difference between two calls to timegm'
-    );
-}
-
-# bugid #19393
-# At a DST transition, the clock skips forward, eg from 01:59:59 to
-# 03:00:00. In this case, 02:00:00 is an invalid time, and should be
-# treated like 03:00:00 rather than 01:00:00 - negative zone offsets used
-# to do the latter
-{
-    my $hour = ( localtime( timelocal( 0, 0, 2, 7, 3, 102 ) ) )[2];
-
-    # testers in US/Pacific should get 3,
-    # other testers should get 2
-    ok( $hour == 2 || $hour == 3, 'hour should be 2 or 3' );
-}
-
-for my $p (@years) {
-    my ( $year, $is_leap_year ) = @$p;
-
-    my $string = $is_leap_year ? 'is' : 'is not';
-    ## no critic (Subroutines::ProtectPrivateSubs)
-    is(
-        Time::Local::_is_leap_year($year), $is_leap_year,
-        "$year $string a leap year"
-    );
-}
+);
 
 SKIP:
 {
     skip 'this platform does not support negative epochs.', 6
```

Comment:
If you're using `subtest`, you could put the `skip` inside the subtest and remove the `SKIP: { ... }` block and indentation using `plan skip_all => 'this platform does not support negative epochs.'`. That way the entire subtest is skipped and no need to keep a count of the tests that are being skipped.

The skip count confused me for a minute, but I did some testing and it's right. I understand why it's right now: To a TAP parser, `subtest` is one test, but Test::Builder sees all the tests inside.

Reply:
Will do this.